lib/seattleflu/db/cli/command/manifest.py

```diff
@@ -238,16 +275,28 @@ def _parse(*,
             })
 
     single_columns = {
+        "collection": collection_column,
+        "kit": kit_column,
         "date": date_column,
         "aliquot_date": aliquot_date_column,
         "test_results": test_results_column,
         "pcr_result": pcr_result_column,
-        "notes": notes_column }
+        "notes": notes_column,
+        "test_strip": test_strip_column,
+        "test_origin": test_origin_column,
+        "arrival_date": arrival_date_column }
 
     for key in single_columns:
         if single_columns[key]:
-            column_map.update({
-                find_one_column(manifest, single_columns[key]): key })
+            map_key = find_one_column(manifest, single_columns[key])
+            # Allow columns to map to multiple keys
+            if map_key in column_map:
+                map_key_copy = map_key + "_copy"
+                manifest[map_key_copy] = manifest[map_key]
+                column_map.update({ map_key_copy: key })
+            else:
+                column_map.update({
+                    find_one_column(manifest, single_columns[key]): key })
```

Comment:
I think there are two issue with this logic:

1. It works for the case of a source column specified twice for two different destination columns, but it doesn't work in the general case for a source column specified three times, for example. The copied column in the column map will be overwritten.

2. There is no check that a column with the `_copy` suffix doesn't already exist, which while unlikely, is definitely possible. Since we don't control the inputs to this code and don't regularly inspect them for changes, we should be as defensive as possible about how we handle them. Breaking in loud ways is better than breaking in silent ways.

I pushed two commits on top of this branch which address both those issues by reworking how the source → destination mapping works. Take a look and see what you think?

Reply:
Yup! Completely agree with the two issues, I think I funneled myself too deep into the specific case for rdt samples. The source → destination mapping looks great!